tests/dCommonTests/LxfmlTests.cpp

```diff
@@ -104,32 +122,131 @@ TEST(LxfmlTests, SplitUsesAllBricksAndNoDuplicates) {
+}
 
-	ASSERT_EQ(originalGroupSet.size(), usedGroupSet.size()) << "Group count mismatch";
-	for (const auto& s : originalGroupSet) ASSERT_NE(usedGroupSet.find(s), usedGroupSet.end()) << "Group missing in splits";
+TEST(LxfmlTests, SplitGroupIssueFile) {
+	// Specific test for the group issue file
+	TestSplitUsesAllBricksAndNoDuplicatesHelper("group_issue.lxfml");
+}
+
+TEST(LxfmlTests, SplitTestFile) {
+	// Specific test for the larger test file
+	TestSplitUsesAllBricksAndNoDuplicatesHelper("test.lxfml");
+}
+
+TEST(LxfmlTests, SplitComplexGroupingFile) {
+	// Test for the complex grouping file - should produce only one split
+	// because all groups are connected via rigid systems
+	std::string data = ReadFile("complex_grouping.lxfml");
+	ASSERT_FALSE(data.empty()) << "Failed to read complex_grouping.lxfml from build directory";
+	
+	std::cout << "\n=== Testing complex grouping file ===" << std::endl;
+	
+	auto results = Lxfml::Split(data);
+	ASSERT_GT(results.size(), 0) << "Split results should not be empty";
+	
+	// The complex grouping file should produce exactly ONE split
+	// because all groups share bricks through rigid systems
+	EXPECT_EQ(results.size(), 1) << "Complex grouping file should produce exactly 1 split (all groups merged)";
+	
+	if (results.size() == 1) {
+		std::cout << "✓ Correctly produced 1 merged split" << std::endl;
+		
+		// Verify the split contains all the expected elements
+		tinyxml2::XMLDocument doc;
+		ASSERT_EQ(doc.Parse(results[0].lxfml.c_str()), tinyxml2::XML_SUCCESS);
+		
+		auto* lxfml = doc.FirstChildElement("LXFML");
+		ASSERT_NE(lxfml, nullptr);
+		
+		// Count bricks
+		int brickCount = 0;
+		if (auto* bricks = lxfml->FirstChildElement("Bricks")) {
+			for (auto* brick = bricks->FirstChildElement("Brick"); brick; brick = brick->NextSiblingElement("Brick")) {
+				brickCount++;
+			}
+		}
+		std::cout << "Contains " << brickCount << " bricks" << std::endl;
+		
+		// Count rigid systems
+		int rigidCount = 0;
+		if (auto* rigidSystems = lxfml->FirstChildElement("RigidSystems")) {
+			for (auto* rs = rigidSystems->FirstChildElement("RigidSystem"); rs; rs = rs->NextSiblingElement("RigidSystem")) {
+				rigidCount++;
+			}
+		}
+		std::cout << "Contains " << rigidCount << " rigid systems" << std::endl;
+		EXPECT_GT(rigidCount, 0) << "Should contain rigid systems";
+		
+		// Count groups
+		int groupCount = 0;
+		if (auto* groupSystems = lxfml->FirstChildElement("GroupSystems")) {
+			for (auto* gs = groupSystems->FirstChildElement("GroupSystem"); gs; gs = gs->NextSiblingElement("GroupSystem")) {
+				for (auto* g = gs->FirstChildElement("Group"); g; g = g->NextSiblingElement("Group")) {
+					groupCount++;
+				}
+			}
+		}
+		std::cout << "Contains " << groupCount << " groups" << std::endl;
+		EXPECT_GT(groupCount, 1) << "Should contain multiple groups (all merged into one split)";
+	} else {
+		std::cout << "✗ Produced " << results.size() << " splits instead of 1" << std::endl;
```

Comment:
Is the test supposed to fail in this case?

Reply:
whoeps